astropy/cosmology/tests/test_core.py

```diff
@@ -0,0 +1,555 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""Testing :mod:`astropy.cosmology.core`."""
+
+##############################################################################
+# IMPORTS
+
+import inspect
+import sys
+import weakref
+from io import StringIO, TextIOWrapper
+from types import MappingProxyType
+
+import pytest
+
+import numpy as np
+
+import astropy.units as u
+from astropy.cosmology import Cosmology, FlatLambdaCDM, Planck18, core
+from astropy.cosmology.core import _COSMOLOGY_CLASSES, Parameter
+
+##############################################################################
+# TESTS
+##############################################################################
+
+
+class TestCosmologyInfo:
+    """Test `astropy.cosmology.core.CosmologyInfo`."""
+
+    @pytest.fixture(params=[None, "", sys.stdout, StringIO])
+    def out(self, request):
+        out = request.param
+        if out is StringIO:
+            out = out()
+
+        return out
+
+    @pytest.fixture
+    def cosmo(self):
+        return Planck18
+
+    @pytest.fixture
+    def cosmo_cls(self):
+        return Planck18.__class__
+
+    @pytest.fixture
+    def info(self, cosmo):
+        # yes, it's a descriptor
+        return cosmo.info
+
+    # ==============================================================
+
+    def test_has_expected_attributes(self, info):
+        """Test ``CosmologyInfo`` has all the expected attributes."""
+        assert info._stats == []
+        assert info.attrs_from_parent == set()
+        assert info.attr_names == {"name", "meta"}
+        assert info._attr_defaults == {'dtype': np.dtype('O')}
+        assert info._attrs_no_copy == set()
+        assert info._info_summary_attrs == ('dtype', 'class')
+        assert info._represent_as_dict_primary_data == "name"
```

Comment:
Seeing it here, are you sure you want to have this as "primary data"? It is really meant for things like `Quantity`, where it is obvious. It is not used for representations or coordinates, where multiple numbers are equally important.

Reply:
Agreed. I've removed ``_represent_as_dict_primary_data``.
But also moved ``info`` to another PR.